### PR TITLE
Sniper rifle buff 2

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

- The sniper rifle is no longer automatic and has to be manually bolted. In exchange the minimum time between fire has been lowered from 4 seconds to 1.5 seconds. (Same as mosin)
- Ammunition now comes from the uplink without any bullets, which have to be brought seperately.
- Sniper rifle rounds will now penetrate structures and mobs, dealing damage on the way (Still blocked by turfs).
- Added EMP sniper rounds
- Added inferno sniper rounds
- Added explosive sniper rounds.
- Adde anti-matter sniper round.

## Why It's Good For The Game

The sniper rifle currently has huge usability issues due to the fact that it has such an insane fire-delay that anyone using it ends up spamming wondering why the gun isn't firing.
This makes the gun bolt-action instead so you have to rack it in order to fire the next shot.
The new ammo types also give the sniper rifle more utility, with some extremly powerful but incredibly expensive ammunition types that may actually invoke fear into enemies rather than a 'oh sniper, better rush' mentality.

## Testing Photographs and Procedure

image

image

image

image

image

image

image

It now has to be chambered similar to how a shotgun works.

## Changelog
:cl:
balance: Sniper rifles now need to be chambered (Like a shotgun, one use to chamber)
balance: Sniper rifles minimum fire delay decreased from 4s to 1.5s.
balance: Sniper rifle ammo is now brought seperately from the magazine.
balance: Added in explosive sniper rounsd.
balance: Added in EMP sniper rounds.
balance: Added in inferno sniper rounds.
balance: Added in anti-matter sniper rounds.
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
Seems interesting and it adds some balance, but i've seen complaints from players(and saw them myself) that some Nukies with thermal + piercing ammo would DESTROY anyone because the piercing ammo would go literally through any type of wall, in fact i remember seeing a HoP on Glow be behind his office, shutters closed and maybe near a reinforced wall and got melted instantly..

Yes, they're piercing rounds but i don't think the should-go meta needs to be thermal+piercing because they don't have a ''collision'' due to... ''piercing'' every type of item, the variety of ammo would certainly add variety but what are your thoughts on that meta?

**Comment by PowerfulBacon** [comment]
I have played X-Ray + Penetrator and I can confirm it is massively over-rated

**Comment by Rukofamicom** [comment]
My only concerns now are the fact that the lavaland syndicates have had their access to ammo dramatically reduced, and that this gun might be excessively powerful now. 

I look forward to seeing how it pans out, and I especially love having it switched to a bolt action for flavor reasons.

**Comment by PowerfulBacon** [comment]
There is no changes to the ammo for lavaland syndies.
The gun comes with 1 full box of ammo, the only thing changed is that the nukie magazine contains no bullets and the bullets come seperately (due to their power and special powers).

**Comment by heepox** [comment]
Syndiekit Bundle_A "sniper" is gonna an ammobox or something if this gets merged.

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
 /obj/item/projectile/bullet/p50
 	name =".50 bullet"
-	speed = 0.4
+	speed = 0.2
 	damage = 70
 	paralyze = 100
 	dismemberment = 50
 	armour_penetration = 50
+	// Will penetrate but damage anything not a wall
+	projectile_piercing = PASSMOB | PASSMACHINE | PASSGLASS | PASSGRILLE | PASSDOORS | PASSFLAPS | PASSSTRUCTURE
```
Comment (Rukofamicom) [inline code/modules/projectiles/projectile/bullets/sniper.dm]:
```suggestion
	projectile_piercing = PASSMOB | PASSMACHINE | PASSTRANSPARENT | PASSGRILLE | PASSDOORS | PASSFLAPS | PASSSTRUCTURE
```
Needs to be updated after the merging of #8518

File: code/modules/projectiles/projectile/bullets/sniper.dm [hunk 2]
```diff
@@ -15,26 +17,13 @@
...
-
 /obj/item/projectile/bullet/p50/penetrator
 	name =".50 penetrator bullet"
 	icon_state = "gauss"
 	damage = 60
-	projectile_piercing = PASSMOB
-	projectile_phasing = (ALL & (~PASSMOB))
+	projectile_piercing = PASSMOB | PASSMACHINE | PASSGLASS | PASSGRILLE | PASSDOORS | PASSFLAPS | PASSSTRUCTURE
+	// Phase directly through everything else
+	projectile_phasing = (ALL & ~(PASSMOB | PASSMACHINE | PASSGLASS | PASSGRILLE | PASSDOORS | PASSFLAPS | PASSSTRUCTURE))
```
Comment (Rukofamicom) [inline code/modules/projectiles/projectile/bullets/sniper.dm]:
```suggestion
	projectile_piercing = PASSMOB | PASSMACHINE | PASSTRANSPARENT | PASSGRILLE | PASSDOORS | PASSFLAPS | PASSSTRUCTURE
	// Phase directly through everything else
	projectile_phasing = (ALL & ~(PASSMOB | PASSMACHINE | PASSTRANSPARENT | PASSGRILLE | PASSDOORS | PASSFLAPS | PASSSTRUCTURE))
```

Same as above

**Review by Rukofamicom — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]
Seems pretty cool, although I worry the antimatter one is a bit too powergamey, 1 hit kill. The 14TC cost is steep though I don't really see anyone purchasing this. Might be cool though.

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -623,7 +623,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 /datum/uplink_item/dangerous/sniper
 	name = "Sniper Rifle"
 	desc = "Ranged fury, Syndicate style. Guaranteed to cause shock and awe or your TC back!"
-	item = /obj/item/gun/ballistic/automatic/sniper_rifle/syndicate
+	item = /obj/item/gun/ballistic/sniper_rifle/syndicate
 	cost = 16
```
Comment (itsmeow) [inline code/modules/uplink/uplink_items.dm:627]: shouldn't the cost of the sniper be lower now that it doesn't come with any ammo?

Comment (Rukofamicom) [reply code/modules/uplink/uplink_items.dm:627]: it does come with ammo still, I had the same misunderstanding with how they had worded things lol

The magazines you buy separately from the gun come with no ammo, but the gun itself still comes with a loaded mag.

**Pull request merged**